### ISSUE TITLE
Tuning two crypted channels: lost lock on channel

### ISSUE THREAD

**Issue by Attrachii** [manio/vdr-plugin-dvbapi#128]
Hi,
i have the problem that i can only watch one encrypted channel with two TV cards. As soon as a second channel is selected, I get the message "lost lock on channel X" in the log and the stream of the first channel stops (see beginning of the log). One encrypted and one unencrypted channel work without problems (see end of log).

**Version**
vdr (2.4.0/2.4.0) - The Video Disk Recorder
dvbapi (2.2.4-GIT-7d51cc4) - SoftCAM for OSCam
streamdev-server (0.6.1-git) - VDR Streaming Server
vnsiserver (1.6.0) - VDR-Network-Streaming-Interface (VNSI) Server

**Log-File**
[log.txt](https://github.com/manio/vdr-plugin-dvbapi/files/2701022/log.txt) is attached.

Sometimes it even happens that the whole VDR crashes and restarts.

**Log-File with crash**
[log-crash.txt](https://github.com/manio/vdr-plugin-dvbapi/files/2701041/log-crash.txt)

I hope you can help me. Thank you in advance.

**Comment by manio** [comment]
No idea what is wrong....
You may try to schedule recordings from two channels at the same time and let it run for some time period... and in the same time don't use a streamdev and vnsi - are the recordings ok?

**Comment by Attrachii** [comment]
Thank you very much for your answer and sorry for waiting so long. 

I have a log here, there I recorded two encrypted channels. unfortunately the same problem occurred:
[log-recording-two-crypted-channels.txt](https://github.com/manio/vdr-plugin-dvbapi/files/2717509/log-recording-two-crypted-channels.txt)

Then i did another test and recorded an encrypted and an unencrypted channel - everything ok. then stopped the unencrypted one and added an encrypted one, unfortunately also the same problem:
[log-recording-one-crypted-one-not-crypted-channel.txt](https://github.com/manio/vdr-plugin-dvbapi/files/2717510/log-recording-one-crypted-one-not-crypted-channel.txt)

Maybe this will help you. If you can't figure it out, I'll switch back to another VDR version and check if it gets better. Thank you for your effort.

**Comment by 9000h** [comment]
@Attrachii 
please try the same with a plain vanilla vdr and dvbapi only.

**Comment by manio** [comment]
Yeah, exactly as @9000h said - please redo the test but only with necessary plugins, which are: -Pdvbapi :)

**Comment by Attrachii** [comment]
After further investigation and spending hours of compiling my own vdr 2.4.0 with the latest patch files, the problem was still there. With all plugins enabled as well with only dvbapi enabled. Then i have decided to install a clean Ubuntu 18.04 LTS with a fresh VDR and plugins from the repository. Only the latest version of dvbapi is custom compiled as it is not available in the repository.

I have now running VDR 2.3.8 and the problem is still there but i have found out a very interresting point: It works totally fine if i look only channels from the same CAID. I use the CAIDs 1830 and 098C. Looking only channels from 1830 is no problem, only channels from 098C is also no problem. But as soon as i start to mix up the CAIDs on the TV cards the problem is right there: One TV card lost the lock on the channel. And its always the channel which i have tuned first. The second channel i have tuned works fine. But even if i stop the second channel the first TV card will not regain the lock on the channel. I had to stop the stream / recording and restart the recording and most of the time that also doesn't help. Only with a restart of the whole VDR all starts to work again. I do now also have the problem that i cannot look one encrypted channel and one unencrypted channel sometimes.

I have done the following test for the encrypted / unencrypted situation: One channel was tuned by VLC on CAID 098C, the other device was a kodi connected via VNSI. I could switch to every 098C channel in kodi as i want and all works fine. If i tuned to an unencrypted channel (Das Erste HD) the VDR cannot get a lock on the channel. As soon as i tuned back to a 098C channel i got a lock on the channel and all works fine.

I have appended a log ([vdr log neu.txt](https://github.com/manio/vdr-plugin-dvbapi/files/2761925/vdr.log.neu.txt)) where only the DVBAPI plugin was loaded and first started a recording of two 1830 channels and all works fine. Then i have stopped one 1830 recording and added a 098C recording and a lost lock on channel happened. After that the VDR has crashed with an emergency exit and restarted.


Sometimes i have also got this error in the syslog:

Jan 15 23:15:38 mediaserver01 kernel: [ 5841.231940] show_signal_msg: 9 callbacks suppressed
Jan 15 23:15:38 mediaserver01 kernel: [ 5841.231946] device 1 receiv[22569]: segfault at 7f76cc0025c0 ip 00007f75f38bbeaa sp 00007f75c37fa4a8 error 4 in libvdr-dvbapi.so.2.3.8[7f75f38ab000+1a000]

Thanks in advance for your help :)
